http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/api/ProxyRoutes.scala

```diff
@@ -262,14 +262,21 @@ class ProxyRoutes(proxyService: ProxyService, corsSupport: CorsSupport, refererC
       pass
     }
 
+  private def requestPath(req: HttpRequest): String = req.uri.toString
+  private val logRequestPath = DebuggingDirectives.logRequest(requestPath _)
+
   private[api] def checkReferer: Directive0 =
     optionalHeaderValueByType(Referer) flatMap {
       case Some(referer) => {
         if (refererConfig.validHosts.contains(referer.uri.authority.toString())) pass
-        else failWith(AuthenticationError())
+        else {
+          logger.info(s"Referer ${referer} is not allowed")
+          logRequestPath.tflatMap(_ => reject)
+        }
       }
       case None => {
-        failWith(AuthenticationError())
+        logger.info(s"Referer header is missing")
+        logRequestPath.tflatMap(_ => reject)
```

Comment:
does `reject` fail with 401?

Reply:
it's probably 40x...I'm assuming this shouldn't matter...

Reply:
sounds good although I would run unit test `ProxyRoutesSpec` locally since the last test in there validates that `StatusCodes.Unauthorized` happens to make sure it still passes

Reply:
actually looks like the test is failing, once it's fixed we should know what status code gets returned

Reply:
it's 405:
- should 401 when Referer is not a valid URI *** FAILED ***
16:15:00 [info]   405 Method Not Allowed did not equal 401 Unauthorized (ProxyRoutesSpec.scala:557)

Reply:
I think we want 401 in this case.. is there a reason to use `reject`? Can we change back to `failWith(AuthenticationError())`?

Reply:
+1 for failing with 401

Reply:
types don't match for `failWith(AuthenticationError())`...there might be a way to make it work..but my IJ is really messed up today, so I can't really figure things out easily...if you guys want, feel free to update the PR....

What's the reason why we want 401 tho? it's not really an auth error either

Reply:
i'll push up the change

Reply:
the types matched up for me and i checked to make sure everything compiles and test passes - maybe it's an issue with your IJ

Reply:
interesting...okay...

still not convinced why 401 is preferred over `reject` tho 🤷‍♀️